extensions/amp-story/1.0/amp-story-consent.js

```diff
@@ -0,0 +1,275 @@
+            },
+            {
+              tag: 'div',
+              attrs: dict({'class': 'i-amphtml-story-consent-content'}),
+              children: [
+                {
+                  tag: 'h3',
+                  attrs: dict({}),
+                  children: [],
+                  unlocalizedString: config.title,
+                },
+                {
+                  tag: 'p',
+                  attrs: dict({}),
+                  children: [],
+                  unlocalizedString: config.message,
+                },
+                {
+                  tag: 'ul',
+                  attrs: dict({'class': 'i-amphtml-story-consent-vendors'}),
+                  children: config.vendors && config.vendors.map(vendor => (
+                    {
+                      tag: 'li',
+                      attrs: dict({'class': 'i-amphtml-story-consent-vendor'}),
+                      children: [],
+                      unlocalizedString: vendor,
+                    })
+                  ),
+                },
+              ],
+            },
+            {
+              tag: 'div',
+              attrs: dict({'class': 'i-amphtml-story-consent-actions'}),
+              children: [
+                {
+                  tag: 'button',
+                  attrs: dict({
+                    'class': 'i-amphtml-story-consent-action ' +
+                        'i-amphtml-story-consent-action-reject',
+                    'on': `tap:${consentId}.reject`,
+                  }),
+                  children: [],
+                  unlocalizedString: 'Decline',
+                },
+                {
+                  tag: 'button',
+                  attrs: dict({
+                    'class': 'i-amphtml-story-consent-action ' +
+                        'i-amphtml-story-consent-action-accept',
+                    'style': config.color ?
+                      `background-color: ${config.color} !important; ` +
+                          `border-color: ${config.color} !important;` : '',
+                    'on': `tap:${consentId}.accept`,
+                  }),
+                  children: [],
+                  unlocalizedString: 'Agree',
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    },
+  ],
+});
+
+/**
+ * The <amp-story-consent> custom element.
+ */
+export class AmpStoryConsent extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @const @private {!../../../src/service/action-impl.ActionService} */
+    this.actions_ = Services.actionServiceForDoc(this.element);
+
+    /** @private {?Object} */
+    this.consentConfig_ = null;
+
+    /** @private {?Element} */
+    this.scrollableEl_ = null;
+
+    /** @private {?Element} */
+    this.storyConsentEl_ = null;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.assertAndParseConfig_();
+
+    const el = this.win.document.querySelector('[publisher-logo-src]');
```

Comment:
I think we should do this the other way; use `closest` from dom.js to search up the tree until we find the `<amp-story>` element, then use that to get its `publisher-logo-src`.  It's not sufficient that `publisher-logo-src` is on any arbitrary tag, and also if we search the whole document, we'll get in trouble once we start allowing more than one story per document.

Reply:
Good point, done